addons/sale/static/src/js/product_configurator_mixin.js

```diff
@@ -440,6 +403,51 @@ var ProductConfiguratorMixin = {
     _getProductId: function ($parent) {
         return parseInt($parent.find('.product_id').val());
     },
+    /**
+     * Returns a debounced `_getCombinationInfo` with a leading and a trailing
+     * call, and which is memoized per `uniqueId`.
+     *
+     * The uniqueId is needed because on the configurator modal there might be
+     * multiple elements triggering the rpc at the same time, and we want each
+     * individual product rpc to be executed, but only once per individual
+     * product.
+     *
+     * The leading execution is to keep good reactivity on the first call, for
```

Comment:
You are correct that this part is only guaranteeing the order of the rpc (to prevent useless rpc), the code to guarantee their order of the return is on the callback function of the rpc, as I said on the other comment.

Reply:
There must be a way to handle this with `web.concurrency`. Not read all the debates around here nor tested the use case but it feels like what you want is a non-debounced event handler with a drop-previous system for the RPCs ?

Reply:
We definitely need a debounced here, because without it, this can spam useless RPC to the point of massively slowing down everything.

Reply:
Only if the user is a skilled crazy person that manages to click on a variant then on another in less than 10ms then on another in less than 10ms etc, right ? :smile:

Reply:
No in this case it's actually necessary:

- the event itself is triggered multiple times due to the mixin being extended in several places, so the exact same RPC would be done 2 or 3 times on every click
- it's a "slow" RPC so a normal user would be fast enough too, it's also triggered on quantity change which is an easy click fast
- not to mention on any click the RPC is done for every optional product so the issue is exacerbated even more

Reply:
>     * the event itself is triggered multiple times due to the mixin being extended in several places, so the exact same RPC would be done 2 or 3 times on every click

Please tell me that this is not true :sob:  Can't this be fixed ? :cry: 

>     * it's a "slow" RPC so a normal user would be fast enough too, it's also triggered on quantity change which is an easy click fast

Why is the RPC for variant changing the same as for quantity changing ? x)

Anyways, what about using `throttle` instead of `debounce`? It has the trailing and leading parameters, less code in stable for nearly what you want. Actually, debouncing or throttling with a high delay (500ms) on click events... I am not sure you will see a lot of differences. The visitor will not continue clicking at a high rate for multiple seconds... and if it does, we can consider that the UI being refreshed every 500ms is a feature, no?

>     * not to mention on any click the RPC is done for every optional product so the issue is exacerbated even more

Also something to fix. If I understand you right, on any click, if there is 100 optional products, 100 rpcs are sent to the server... and in fact even 300 if your point 1 is right ? x) Well... what ? :p

Reply:
We now have throttle instead of the custom debounce, and also we have DropPrevious.

The RPC for the variant change is done on quantity change too because it computes the price (pricelist rules), we can't change that.

I finally fixed the issue with the duplicate events!

There are still as many RPC as there are optional products, I will not change that in the current PR. We could imagine creating a route taking all the combinations to compute as parameter, but we would have the same perf issue as the shop page before, we just can't support many options either way.